File: web/paystack_interop.js
function paystackPopUp(pkTest, email, amount, ref, onClosed, callback) {
  let handler = PaystackPop.setup({
    key: pkTest, // Replace with your public key
    email: email,
    amount: amount,
    ref: ref, // generates a pseudo-unique reference. Please replace with a reference you generated. Or remove the line entirely so our API will generate one for you
    onClose: function () {
      alert("Window closed.");
      onClosed();
    },
    callback: function (response) {
      callback();
      let message = "Payment complete! Reference: " + response.reference;
      alert(message);
    },
  });
  return handler.openIframe();
}

File: lib/services/paystack_interop.dart
import 'package:js/js.dart';

@JS()
external paystackPopUp(
  String pkTest,
  String email,
  String amount,
  String ref,
  void Function() onClosed,
  void Function() callback,
);



















// @JS('PaystackPop')
// library facebook_integration.js;

// import 'package:js/js.dart';

// typedef PaystackCallback = void Function(dynamic response);

// @JS('setup')
// external setup(SetUpOptions options);

// @JS('openIframe')
// external openIframe();

// @JS()
// @anonymous
// class SetUpOptions {
//   external factory SetUpOptions({
//     String key,
//     String email,
//     int amount,
//     String ref,
//     PaystackCallback onClosed,
//     PaystackCallback callback,
//   });
//   external String key;
//   external String email;
//   external int amount;
//   external String ref;
//   external PaystackCallback onClosed;
//   external PaystackCallback callback;
// }
